### PR TITLE
Adding trace logs to LDAP code to log the ldap query results

### PR DESCRIPTION
**Pull request by prachidamle**
We need more logs to log the data coming in from LDAP, logging at TRACE.

### CODE REVIEW

**Comment by prachidamle** [comment]
@ibuildthecloud @alena1108 can you merge this for pre-release this week

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java [hunk 1]
```diff
@@ -48,6 +50,8 @@
     public static final String RESTRICTED_ACCESSMODE = "restricted";
     public static final String UNRESTRICTED_ACCESSMODE = "unrestricted";
 
+    private static final Logger logger = LoggerFactory.getLogger(AbstractTokenUtil.class);
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java]: Can you rename to `log`

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java [hunk 2]
```diff
@@ -178,7 +183,12 @@ protected boolean isAllowed(Map<String, Object> jsonData) {
         }
         List<String> idList = (List<String>) jsonData.get(ID_LIST);
         for (String id : idList) {
-            identities.add(Identity.fromId(id));
+            Identity identityObj = Identity.fromId(id);
+            if (identityObj != null) {
+                identities.add(identityObj);
+            } else {
+                logger.trace("Identity is null for id: " + id);
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java]: Can you use slf4j format `log.trace("Identity is null for id: {}",id);`

Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java]: Same with all log lines below

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java [hunk 3]
```diff
@@ -106,6 +106,7 @@ private SearchResult userRecord(String query) {
             SearchResult result;
             try {
                 result = results.next();
+                logger.trace("Found  LDAP SearchResult object: " + result.toString());
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java]: Don't call `toString()`

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java [hunk 4]
```diff
@@ -128,7 +129,8 @@ private SearchResult userRecord(String query) {
     private Set<Identity> getIdentities(SearchResult result) {
         final Set<Identity> identities = new HashSet<>();
         final Attribute memberOf = result.getAttributes().get(getConstantsConfig().getUserMemberAttribute());
-
+        logger.trace("ADConstants userMemberAttribute() " + getConstantsConfig().getUserMemberAttribute());
+        logger.trace("SearchResult memberOf attribute " + memberOf.toString());
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java]: Don't call `toString()`

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java [hunk 5]
```diff
@@ -211,6 +219,28 @@ protected Identity attributesToIdentity(LdapName dn){
         }
     }
 
+    private String removeSpacesAroundRDN(String dn) {
+
+        if (dn == null || StringUtils.isEmpty(dn)) {
```
Comment (alena1108) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java]: I believe stringUtils.isEmpty returns true for null string. So the first check condition can be skipped

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java (hunk 1 above)
Comment (prachidamle) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java]: renamed to log

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java (hunk 2 above)
Comment (prachidamle) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java]: Made the changes

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java (hunk 3 above)
Comment (prachidamle) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java]: removed.

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java (hunk 4 above)
Comment (prachidamle) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java]: removed this

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java (hunk 5 above)
Comment (prachidamle) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java]: Correct, removed that check

**Pull request merged**